### PR TITLE
(BKR-321) Added lxc driver/hypervisor for beaker

### PR DESCRIPTION
**Pull request by ashish1099**
#843 continued here.
I will update the ticket.

### CODE REVIEW

**Comment by anodelman** [comment]
Can we get some spec tests here to ensure that it is generating expected output?  Especially since we do not have the ability to test this internally.

**Comment by anodelman** [comment]
Also, please yard doc your new methods.

**Comment by anodelman** [comment]
Last comments still current.

**Comment by ashish1099** [comment]
@anodelman 
https://gist.github.com/ashish1099/13a8442621b383a6eab9

I will write up the docs,

**Comment by anodelman** [comment]
Thanks, I'll wait on the yard docs before merging.

File: lib/beaker/hypervisor/lxc.rb [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            puts `yum install -y sudo initscripts openssh-server openssh-clients #{Beaker::HostPrebuiltSteps::UNIX_PACKAGES.join(' ')}`
+            puts `service sshd start`
+          when /opensuse/, /sles/
+            puts `zypper -n in openssh #{Beaker::HostPrebuiltSteps::SLES_PACKAGES.join(' ')}`
+            puts `ssh-keygen -t rsa -f /etc/ssh/ssh_host_rsa_key`
+            puts `ssh-keygen -t dsa -f /etc/ssh/ssh_host_dsa_key`
+            puts `sed -ri 's/^#?UsePAM .*/UsePAM no/' /etc/ssh/sshd_config`
+          else
+            # TODO add more platform steps here
+            raise "platform #{host['platform']} not yet supported on docker"
```
Comment (electrical) [inline lib/beaker/hypervisor/lxc.rb]: s/docker/lxc/  ?

**Comment by anodelman** [comment]
Waiting on yard docs.

**Comment by kevpl** [comment]
last comment still current

**Comment by ashish1099** [comment]
I have generated the yard document, I'm not sure where to put it. Let me know. I have got it ready

**Comment by kevpl** [comment]
@ashish1099 yard doc comments live just above the method that they're documenting, so you should put those comments there.  Let me know if you have any questions about this.

Also, it appears that a rebase has gone badly here.  You'll have to clean up the commit history for this PR  in order for it to be merged.

**Comment by anodelman** [comment]
Last review comment still current.

**Comment by kevpl** [comment]
last review comment still current

**Comment by kevpl** [comment]
last review comment still current

**Comment by anodelman** [comment]
No activity in weeks and now requires a rebase.  Closing out.  Please update the re-open if you still want this functionality.

**Pull request closed without merging**